pep-0636.rst

```diff
@@ -328,56 +269,150 @@ The patterns we have explored above can do some powerful data filtering, but som
+                ui.play(url)
+            case {"sound": _, "format": _}
+                warning("Unsupported audio format")
+
+The keys in your mapping pattern need to be literals, but the values can be any
+pattern. As in sequence patterns, all subpatterns have to match for the general
+pattern to match. 
+
+You can use ``**rest`` within a mapping pattern to capture additional keys in
+the subject. Note that if you omit this, extra keys in the subject will be
+ignored while matching, i.e. the message 
+``{"text": "foo", "color": "red", "style": "bold"}`` will match the first pattern
+in the example above. 
 
 Matching objects
 ----------------
 
-UI events motivations. describe events in dataclasses. inspiration for event objects
-can be taken from https://www.pygame.org/docs/ref/event.html
+Our adventure is being a success and we have been asked to implement a graphical
+interface. Or UI toolkit of choice allows us to write an event loop where we can get a new
+event object by calling ``event.get()``. The resulting object can have different type and
+attributes according to the user action, for example:
+
+* A ``KeyPress`` object is generated when the user presses a key. It has a ``key_name``
+  attribute with the name of the key pressed, and some other attributes regarding modifiers
+* A ``Click`` object is generated when the user clicks the mouse. It has an attribute
+  position with the coordinates of the pointer.
+* A ``Quit`` object is generated when the user clicks on the close button for the game
+  window.
+
+Rather than writing multiple ``isinstance()`` checks, we can use patterns to recognize
+different kinds of objects, and also apply patterns to its attributes::
+
+    match event.get():
+        case Click(position=(x, y)):
+            handle_click_at(x, y)
+        case KeyPress(key_name="Q") | Quit():
+            game.quit()
+        case KeyPress(key_name="up arrow"):
+            game.go_north()
+        ...
+        case KeyPress():
+            pass # Ignore other keystrokes 
+        case other_event:
+            raise ValueError(f"Unrecognized event: {other_event}")
+
+A pattern like ``Click(position=(x, y))`` only matches if the actual event is a subclass of
+the ``Click`` class. It will also requires that the event has a ``position`` attribute
+that matches the ``(x, y)`` pattern. If there's a match, the locals ``x`` and ``y`` will
+get the expected values.
+
+A pattern like ``KeyPress()``, with no arguments will match any object which is an
+instance of the ``KeyPress`` class. Only the attributes you specify in the pattern are
+matched, and any other attributes are ignored.
+
+Matching positional attributes
+------------------------------
+
+The previous section described how to match named attributes when doing an object match.
+For some objects it could be desirable to describe the matched arguments positionally
+(especially if there are very few attributes to match that have a "standard" ordering).
```

Comment:
```suggestion
For some objects it could be convenient to describe the matched arguments by position
(especially if there are only a few attributes and they have a "standard" ordering).
```